plugins/cassandra/cassandraembed.cpp

```diff
@@ -3325,54 +3638,59 @@ class CCasssandraWorkUnitFactory : public CWorkUnitFactory, implements ICassandr
             if (opt && val)
             {
-                VStringBuffer optstr("%s=%s", opt, val);
-                options.append(optstr);
+                if (strieq(opt, "randomWuidSuffix"))
+                    randomizeSuffix = atoi(val);
+                else if (strieq(opt, "traceLevel"))
+                    traceLevel = atoi(val);
+                else
+                {
+                    VStringBuffer optstr("%s=%s", opt, val);
+                    options.append(optstr);
+                }
             }
         }
         cluster.setOptions(options);
-        session.set(cass_session_new());
-//        CassandraFuture future(cass_session_connect_keyspace(session, cluster, "hpcc"));
-        CassandraFuture future(cass_session_connect(session, cluster));  // Since we don't know if HPCC keyspace exists, easier not to connect to it but rather specify it explicitly in each query
-        future.wait("connect");
-        VStringBuffer create("CREATE KEYSPACE IF NOT EXISTS hpcc WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1' } ;"); // MORE - options from props!
-        executeSimpleCommand(session, create);
-        ensureTable(session, workunitsMappings);
-        ensureTable(session, wuResultsMappings);
-        ensureTable(session, wuVariablesMappings);
-        ensureTable(session, wuExceptionsMappings);
-        ensureTable(session, wuStatisticsMappings);
+        if (cluster.keyspace.isEmpty())
+            cluster.keyspace.set("hpcc");
+        connect();
     }
+
     ~CCasssandraWorkUnitFactory()
     {
     }
-    /*
-    virtual IWorkUnit * createWorkUnit(const char * app, const char * user, ISecManager *secmgr, ISecUser *secuser) { printStackReport(); UNIMPLEMENTED; }
-    virtual bool deleteWorkUnit(const char * wuid, ISecManager *secmgr, ISecUser *secuser) { UNIMPLEMENTED; }
-    virtual IConstWorkUnit * openWorkUnit(const char * wuid, bool lock, ISecManager *secmgr, ISecUser *secuser)
-    {
-        // MORE - what to do about lock?
-        Owned<IPTree> wuXML = cassandraToWorkunitXML(session, wuid);
-        if (wuXML)
-            return new CConstCassandraWorkUnit(wuXML.getClear(), NULL, NULL);
-        else
-            return NULL;
-    }
-    virtual IWorkUnit * updateWorkUnit(const char * wuid, ISecManager *secmgr, ISecUser *secuser) { UNIMPLEMENTED; }
-    virtual int setTracingLevel(int newlevel) { UNIMPLEMENTED; }
-    virtual IWorkUnit * createNamedWorkUnit(const char * wuid, const char * app, const char *user, ISecManager *secmgr, ISecUser *secuser) { UNIMPLEMENTED; }
-*/
+
     virtual CLocalWorkUnit* _createWorkUnit(const char *wuid, ISecManager *secmgr, ISecUser *secuser)
     {
-        unsigned suffix = 0;
+        unsigned suffix;
+        unsigned suffixLength;
+        if (randomizeSuffix)  // May need to enable this option if you are expecting to create hundreds of workunits / second
+        {
+            suffix = rand();
```

Comment:
Only if multithreaded access to it crashes (rather than just being sub-optimal in terms of randomness).

Is there a better random number source I can use?